File: challenge.js
function solution(string) {
  let piece;

  for (let i = 1; i < string.length; i++) {

    if (string[i] === string[0]) {
      piece = string.slice(0, i);
      const regExp = new RegExp(piece, "g");
      const match = string.match(regExp);

      if (match.join("") === string) {
        return match.length;
      }
    }
  }
  return 0
}

File: challenge.py
from logging import StringTemplateStyle
import re

def solution(s):
  piece = ''
  for x in range(1, len(s), 1):
    if s[x] == s[0]:
      piece = s[0:x]
      allPieces = re.findall(piece, s)
      if ''.join(allPieces) == s:
        return len(allPieces)
  return 0



solution('abcabcabcabc')
solution('abccbaabccba')